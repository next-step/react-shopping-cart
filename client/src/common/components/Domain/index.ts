export { default as CartItem } from './CartItem';
export { default as MyOrderListItem } from './Order/MyOrderList/Item';
export { default as MyOrderListHeader } from './Order/MyOrderList/Header';

export { default as OrderdItem } from './Order/OrdredItem/OrderdItem';
export { default as Payment } from './Payment';
export { default as ProductItem } from './Product/Item';
export { default as ProductDetailItem } from './Product/DetailItem';
