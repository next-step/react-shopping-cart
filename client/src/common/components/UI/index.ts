export { default as Button } from './Button';
export { default as Dialog } from './Dialog';
export { default as ErrorMessage } from './ErrorMessage';
export { default as HorizontalLine } from './HorizontalLine';
export { default as Image } from './Image';
export { default as Input } from './Input';
export { default as NavBar } from './NavBar';
export { default as PageHeader } from './PageHeader';
export { default as Pagination } from './Pagination';
export { default as Spinner } from './Spinner';
