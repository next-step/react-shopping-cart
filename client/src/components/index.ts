export { default as Checkbox } from './Checkbox';
export { default as LoadingIcon } from './LoadingIcon';
export { default as Button } from './Button';
export { default as Counter } from './Counter';
