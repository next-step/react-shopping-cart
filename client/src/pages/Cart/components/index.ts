export { default as CartItem } from './CartItem';
export { default as CartList } from './CartList';
export { default as CartListFallback } from './CartList/Skeleton';
