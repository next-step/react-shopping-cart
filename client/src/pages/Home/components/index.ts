export { default as ProductList } from './ProductList';
export { default as Product } from './Product';
