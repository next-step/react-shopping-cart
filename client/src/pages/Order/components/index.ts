export { default as OrderList } from './OrderList';
export { default as OrderListFallback } from './OrderList/Skeleton';
