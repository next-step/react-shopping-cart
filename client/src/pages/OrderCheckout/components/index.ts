export { default as OrderCheckoutItem } from './OrderCheckoutItem';
export { default as OrderCheckoutList } from './OrderCheckoutList';
export { default as OrderCheckoutListFallback } from './OrderCheckoutList/Skeleton';
