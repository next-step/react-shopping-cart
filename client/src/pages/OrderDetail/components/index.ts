export { default as OrderDetailList } from './OrderDetailList';
export { default as OrderDetailListFallback } from './OrderDetailList/Skeleton';
