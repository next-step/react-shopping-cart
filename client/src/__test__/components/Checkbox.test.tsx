import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { Checkbox } from 'components';

describe('Checkbox 컴포넌트', () => {
  test('Checkbox 컴포넌트 렌더링', () => {
    render(<Checkbox />);

    const checkbox = screen.getByRole('checkbox');

    expect(checkbox).toBeDefined();
  });

  test('Checkbox 컴포넌트의 초기 상태는 체크되지 않은 상태여야 한다.', () => {
    render(<Checkbox />);

    const checkbox = screen.getByRole('checkbox');

    expect(checkbox).not.toBeChecked();
  });

  test('checked prop을 전달할 경우 체크된 상태여야 한다.', () => {
    const handleChange = jest.fn();
    render(<Checkbox checked onChange={handleChange} />);

    const checkbox = screen.getByRole('checkbox');

    expect(checkbox).toBeChecked();
  });

  test('체크박스를 클릭할 경우 체크박스의 체크 상태가 변경되어야 한다.', () => {
    const handleChange = jest.fn();
    render(<Checkbox onChange={handleChange} />);

    const checkbox = screen.getByRole('checkbox');

    expect(checkbox).not.toBeChecked();

    fireEvent.click(checkbox);
    expect(handleChange).toBeCalled();
    expect(checkbox).toBeChecked();

    fireEvent.click(checkbox);
    expect(handleChange).toBeCalled();
    expect(checkbox).not.toBeChecked();
  });

  /**
   * @see https://github.com/testing-library/dom-testing-library/issues/92
   * fireEvent의 click으로 input checkbox의 disabled를 테스팅할 수 없음
   */
  test('disabled 상태의 체크박스는 체크 상태를 변경할 수 없어야 한다.', () => {
    const handleChange = jest.fn();
    render(<Checkbox onChange={handleChange} disabled />);

    const checkbox = screen.getByRole('checkbox');

    expect(checkbox).toHaveAttribute('disabled');
    expect(checkbox).not.toBeChecked();

    userEvent.click(checkbox);

    expect(checkbox).toHaveAttribute('disabled');
    expect(checkbox).not.toBeChecked();
  });

  test('size prop 중 large snapshot', () => {
    render(<Checkbox size="large">Large</Checkbox>);
    const checkbox = screen.getByRole('checkbox');
    expect(checkbox).toMatchSnapshot();
  });

  test('size prop 중 medium snapshot', () => {
    render(<Checkbox size="medium">Medium</Checkbox>);
    const checkbox = screen.getByRole('checkbox');
    expect(checkbox).toMatchSnapshot();
  });

  test('size prop 중 small snapshot', () => {
    render(<Checkbox size="small">Small</Checkbox>);
    const checkbox = screen.getByRole('checkbox');
    expect(checkbox).toMatchSnapshot();
  });
});
