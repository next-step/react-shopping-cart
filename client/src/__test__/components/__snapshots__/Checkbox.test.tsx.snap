// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox 컴포넌트 size prop 중 large snapshot 1`] = `
<input
  class="checkbox css-3p8rgi"
  id=":r5:"
  type="checkbox"
/>
`;

exports[`Checkbox 컴포넌트 size prop 중 medium snapshot 1`] = `
<input
  class="checkbox css-j5ofk0"
  id=":r6:"
  type="checkbox"
/>
`;

exports[`Checkbox 컴포넌트 size prop 중 small snapshot 1`] = `
<input
  class="checkbox css-16zo3cc"
  id=":r7:"
  type="checkbox"
/>
`;
